apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java

```diff
@@ -406,18 +407,19 @@ static Extension createExtensionDate(
       CcwCodebookInterface ccwVariable, Optional<BigDecimal> dateYear) {
 
     Extension extension = null;
+    if (!dateYear.isPresent()) {
+      throw new NoSuchElementException();
+    }
     try {
-      String stringDate = dateYear.get().toString() + "-01-01";
-      Date date1 = new SimpleDateFormat("yyyy-MM-dd").parse(stringDate);
-      DateType dateYearValue = new DateType(date1, TemporalPrecisionEnum.YEAR);
+      String stringDate = String.format("%0" + 4 + "d", dateYear.get().intValue());
```

Comment:
I'm not sure of the value of concatenating here. I think that format("%04d"...) is pretty typical and easier to make sense of at a glance.

Reply:
Good catch. I initially was thinking of creating a variable for digit length hence concatenation, but since the year is fixed at 4, there's no need.